### PR TITLE
utils_net: Update to start openvswitch if needed

### PR DESCRIPTION
**Pull request by Yingshun**
If the openvswitch service is not running on the host, we should
start it before testing.

Signed-off-by: Yingshun Cui <yicui@redhat.com>

### CODE REVIEW

**Comment by Yingshun** [comment]
**Test result:**
```

2020-10-19 05:25:02,398 process          L0626 INFO | Running 'systemctl status openvswitch'
2020-10-19 05:25:02,425 process          L0458 DEBUG| [stdout] * openvswitch.service - Open vSwitch
2020-10-19 05:25:02,426 process          L0714 INFO | Command 'systemctl status openvswitch' finished with 3 after 0.020351171493530273s
2020-10-19 05:25:02,427 process          L0458 DEBUG| [stdout]    Loaded: loaded (/usr/lib/systemd/system/openvswitch.service; disabled; vendor preset: disabled)
2020-10-19 05:25:02,427 process          L0458 DEBUG| [stdout]    Active: inactive (dead) (thawing) since Mon 2020-10-19 05:24:46 EDT; 15s ago
2020-10-19 05:25:02,428 process          L0458 DEBUG| [stdout]   Process: 91294 ExecStop=/bin/true (code=exited, status=0/SUCCESS)
2020-10-19 05:25:02,428 process          L0458 DEBUG| [stdout]  Main PID: 78494 (code=exited, status=0/SUCCESS)
2020-10-19 05:25:02,428 process          L0458 DEBUG| [stdout]
... ... 
systemd[1]: Stopped Open vSwitch.
2020-10-19 05:25:02,455 service          L0472 DEBUG| Setting ignore_status to True.
2020-10-19 05:25:02,456 process          L0626 INFO | Running 'systemctl start openvswitch'
2020-10-19 05:25:02,964 process          L0714 INFO | Command 'systemctl start openvswitch' finished with 0 after 0.50032639503479s

```

**Comment by Yingshun** [comment]
It's introduced by 2e361394efde1468f.

**Review by kylazhang — APPROVED** [review]

File: virttest/utils_net.py
Comment (pevogam) [inline virttest/utils_net.py:4266]: Just wondering, is there a way to achieve the same without using the a remote runner? This class was planned to be deprecated and I am not sure what is it that it provides here that a normal remote session won't provide.

Comment (Yingshun) [reply virttest/utils_net.py:4266]: No, I think RemoteRunner.run() is the only way to manage services with session via create_service() for now. It requires a process.CmdResult instance, so if you want to remove virttest.remote.RemoteRunner, maybe need to update service.FactoryHelper() or add some new one.

BTW, may I ask why you want to deprecate RemoteRunner? We have 11 files in tp-libvirt which call that, some cases will be impacted if you do that. 
So I wonder if we can get more information about the virttest.remote changes. Is there an issue or something to track the changes?

Comment (Yingshun) [reply virttest/utils_net.py:4266]: @pevogam  Would you please review it again? Thanks a lot!

Comment (pevogam) [reply virttest/utils_net.py:4266]: Hi, sorry I somehow missed your last comment.

> No, I think RemoteRunner.run() is the only way to manage services with session via create_service() for now.

If you mean system services than we are just adding a [utility](https://github.com/avocado-framework/aexpect/pull/54/files#diff-a5831b347e80eb0db7ab3201c357ec5244c2d053d3641457aefe6db58c219147R515) doing this as well (I guess for an unrelated reason).

> BTW, may I ask why you want to deprecate RemoteRunner? We have 11 files in tp-libvirt which call that, some cases will be impacted if you do that.

If we go anywhere near the plan we will make sure to make everyone know and deprecate it slowly so don't expect any breakage any time soon. That doesn't mean however that there aren't better options to move to even right now.

> So I wonder if we can get more information about the virttest.remote changes. Is there an issue or something to track the changes?

The basic PR where the `remote` module was moved to aexpect (where it belongs together with all session handling) was the one where we first wondered how many people are using the RemoteRunner and decided to keep it as a leftover utility in VT that we might one day deprecate.

In your particular case I still don't see anything special about using the RemoteRunner other than the additional `CmdResult` which could either be added as a shared wrapper of a session result or the API of a few places could simply be adapted to use a command status and output as two separate variables.

Comment (Yingshun) [reply virttest/utils_net.py:4266]: @pevogam Thanks for you reply! I understand your concerns, but I'm wondering if we can merge this pr if the schedule of deprecating RemoteRunner is not fixed.  It has been created for one month and a high priority case's failed by it.

@dzhengfy I know you have a lot of experience in remote session operations, could you please share your thought?

**Review by pevogam — APPROVED** [review]
Hi @Yingshun, this looks good. As long as you are acknowledging that support for the remote runner might not be forever and are clear on the alternatives we should look at in the future, I am not going to block this change.

**Comment by Yingshun** [comment]
> Hi @Yingshun, this looks good. As long as you are acknowledging that support for the remote runner might not be forever and are clear on the alternatives we should look at in the future, I am not going to block this change.

Yes, got it. Thank you @pevogam !

**Pull request merged**
